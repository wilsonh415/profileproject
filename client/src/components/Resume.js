import React from 'react';
import { Card } from 'react-bootstrap';
import cplogo from '../images/cplogo.jpeg';
import lowelllogo from '../images/lowelllogo.png';
import llnllogo from '../images/llnllogo.jpeg';
import sfologo from '../images/sfologo.png';
 
class Resume extends React.Component {
 
   render() {
       const educationStyle = {
           top: "10vh",
           marginLeft: "30vw",
           position: "absolute",
           fontFamily: "Georgia"
       }
       const experienceStyle = {
           top: "77vh",
           marginLeft: "30vw",
           position: "absolute",
           fontFamily: "Georgia"
       }
       const edustyles = {
           top: "18vh",
           position: "absolute",
           marginLeft: "30vw",
       };
       const expstyles = {
           top: "85vh",
           position: "absolute",
           marginLeft: "30vw"
       };
       const cardStyle = {
           width: "37vw",
           height: "22vh",
           backgroundColor: "white",
           display: "flex",
           marginBottom: "5vh",
           borderRadius: "35px"
       };
       const cardImgStyle = {
           marginLeft: "10px",
           marginTop: "10px",
           width: "120px",
           height: "120px"
       };
       const llnlStyle = {
           marginLeft: "27px",
           marginTop: "35px",
           width: "84px",
           height: "84px"
       };
       const sfoStyle = {
           marginLeft: "20px",
           marginTop: "10px",
           width: "100px",
           height: "120px"
       };
       const subTextStyle = {
           fontFamily: "Raleway",
           fontSize: "15px",
           float: "left"
       };
       const llnlSubText = {
           fontFamily: "Raleway",
           fontSize: "15px",
           float: "left",
           marginLeft: "22px"
       };
       const sfoSubText = {
           fontFamily: "Raleway",
           fontSize: "15px",
           float: "left",
           marginLeft: "14px"
       };
       const titleStyle = {
           fontFamily: "Georgia",
           fontSize: "20px",
           float: "left",
           marginTop: "1.5vh"
       };
       return (
           <div>
               <h2 style={educationStyle}>
                   <b>Education</b>
               </h2>
               <div style={edustyles}>
                   <Card style={cardStyle}>
                       <div style={{display: "flex"}}>
                           <Card.Img variant="left" src={cplogo} style={cardImgStyle}/>
                           <Card.Body>
                               <Card.Title style={titleStyle}>
                                   <b>California Polytechnic State <br/>University, San Luis Obispo</b>
                               </Card.Title>
                               <Card.Text style={subTextStyle}>
                                   San Luis Obispo, CA &nbsp; <b>|</b> &nbsp; Sep 2017 - Present
                               </Card.Text>
                               <Card.Text style={subTextStyle}>
                                   B.S. in Computer Science
                               </Card.Text>
                           </Card.Body>
                       </div>
                   </Card>
                   <Card style={cardStyle}>
                       <div style={{display: "flex"}}>
                           <Card.Img variant="left" src={lowelllogo} style={cardImgStyle}/>
                           <Card.Body>
                               <Card.Title style={titleStyle}>
                                   <b>Lowell High School</b>
                               </Card.Title>
                               <Card.Text style={subTextStyle}>
                                   San Francisco, CA &nbsp; <b>|</b> &nbsp; Aug 2013 - May 2017
                               </Card.Text>
                               <Card.Text style={subTextStyle}>
                                   11 Advanced Placement Classes
                               </Card.Text>
                           </Card.Body>
                       </div>
                   </Card>
               </div>
 
               <h2 style={experienceStyle}>
                   <b>Work Experience</b>
               </h2>
               <div style={expstyles}>
               <Card style={cardStyle}>
                       <div style={{display: "flex"}}>
                           <Card.Img variant="left" src={llnllogo} style={llnlStyle}/>
                           <Card.Body>
                               <Card.Title style={{fontFamily: "Georgia", fontSize: "20px", marginLeft: "-30px", marginTop: "1.5vh"}}>
                                   <b>Lawrence Livermore National Laboratory</b>
                               </Card.Title>
                               <Card.Text style={llnlSubText}>
                                   Livermore, CA &nbsp; <b>|</b> &nbsp; Apr 2019 - Aug 2019
                               </Card.Text>
                               <Card.Text style={llnlSubText}>
                                   Computation Student Scholar
                               </Card.Text>
                           </Card.Body>
                       </div>
                   </Card>
                   <Card style={cardStyle}>
                       <div style={{display: "flex"}}>
                           <Card.Img variant="left" src={sfologo} style={sfoStyle}/>
                           <Card.Body>
                               <Card.Title style={{fontFamily: "Georgia", fontSize: "20px", marginLeft: "-30px", marginTop: "1.5vh"}}>
                                   <b>City & County of San Francisco <br/> (SFO Airport - ITT Department)</b>
                               </Card.Title>
                               <Card.Text style={sfoSubText}>
                                   San Francisco, CA &nbsp; <b>|</b> &nbsp; Jun 2017 - Aug 2017
                               </Card.Text>
                               <Card.Text style={sfoSubText}>
                                   Public Service Trainee (Software Engineering Intern)
                               </Card.Text>
                           </Card.Body>
                       </div>
                   </Card>
               </div>
           </div>
       );
   }
}
 
export default Resume;
