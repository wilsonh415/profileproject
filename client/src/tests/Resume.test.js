import React from 'react';
import { shallow } from 'enzyme';
import { configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Resume from '../components/Resume';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });


test('Resume component renders correctly', () => {
    const component = renderer.create(<Resume></Resume>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});