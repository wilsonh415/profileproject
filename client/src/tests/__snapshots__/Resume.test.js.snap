// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume component renders correctly 1`] = `
<div
  className="resumeStyle"
>
  <h2
    style={
      Object {
        "backgroundColor": "skyblue",
        "clear": "both",
        "float": "left",
        "fontFamily": "Georgia",
        "fontSize": "30px",
        "marginLeft": "6vw",
        "position": "relative",
        "top": "3vh",
      }
    }
  >
    <b>
      Education
    </b>
  </h2>
  <div
    style={
      Object {
        "clear": "both",
        "marginLeft": "5vw",
        "position": "relative",
        "top": "5vh",
        "width": "44vw",
      }
    }
  >
    <div
      className="accordion"
    >
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": "35px",
            "flexDirection": "row",
            "float": "left",
            "marginBottom": "5vh",
            "maxHeight": "28vh",
            "minHeight": "22vh",
            "minWidth": "38vw",
          }
        }
      >
        <img
          className="card-img-left"
          src="cplogo.jpeg"
          style={
            Object {
              "height": "120px",
              "marginLeft": "10px",
              "marginRight": "15px",
              "marginTop": "10px",
              "width": "120px",
            }
          }
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
            style={
              Object {
                "float": "left",
                "fontFamily": "Georgia",
                "fontSize": "20px",
              }
            }
          >
            <a
              href="https://www.calpoly.edu/"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
              target="_blank"
            >
              <b>
                California Polytechnic State 
                <br />
                University, San Luis Obispo
              </b>
            </a>
          </div>
          <br />
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
              }
            }
          >
              San Luis Obispo, CA   
            <b>
              |
            </b>
               Sep 2017 - Present
          </p>
          <br />
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
              }
            }
          >
              B.S. in Computer Science
          </p>
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <b>
            &gt;
          </b>
        </button>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            <div
              style={
                Object {
                  "float": "left",
                }
              }
            >
              <b>
                <u>
                  Relevant Coursework:
                </u>
              </b>
              <br />
              <ul
                style={
                  Object {
                    "fontSize": "9px",
                    "textAlign": "left",
                  }
                }
              >
                <li>
                  Design & Implementation of Databases
                </li>
                <li>
                  Design & Analysis of Algorithms
                </li>
                <li>
                  Theory of Computation
                </li>
                <li>
                  Operating Systems
                </li>
                <li>
                  Statistics with R
                </li>
                <li>
                  User-Centered Interface & Design
                </li>
                <li>
                  Human Computer Interaction
                </li>
                <li>
                  OOP & Data Structures
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginLeft": "5vw",
        "position": "relative",
        "top": "5vh",
        "width": "44vw",
      }
    }
  >
    <div
      className="accordion"
    >
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": "35px",
            "flexDirection": "row",
            "float": "left",
            "marginBottom": "5vh",
            "maxHeight": "28vh",
            "minHeight": "22vh",
            "minWidth": "38vw",
          }
        }
      >
        <img
          className="card-img-left"
          src="lowelllogo.png"
          style={
            Object {
              "height": "100px",
              "marginLeft": "20px",
              "marginRight": "25px",
              "marginTop": "20px",
              "width": "100px",
            }
          }
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
            style={
              Object {
                "float": "left",
                "fontFamily": "Georgia",
                "fontSize": "20px",
              }
            }
          >
            <a
              href="https://en.wikipedia.org/wiki/Lowell_High_School_(San_Francisco)"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
              target="_blank"
            >
              <b>
                Lowell High School
              </b>
            </a>
          </div>
          <br />
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
              }
            }
          >
            San Francisco, CA   
            <b>
              |
            </b>
               Aug 2013 - May 2017
          </p>
          <br />
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
              }
            }
          >
            11 Advanced Placement Classes
          </p>
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <b>
            &gt;
          </b>
        </button>
        <div
          aria-expanded={null}
          className="collapse"
        >
          <div
            className="card-body"
          >
            <div>
              <p
                style={
                  Object {
                    "fontSize": "9px",
                  }
                }
              >
                Lowell High School is a selective 
                <br />
                public magnet school where admission
                <br />
                tests are necessary for enrollment.
              </p>
              <b>
                <u>
                  Relevant Coursework:
                </u>
              </b>
              <ul
                style={
                  Object {
                    "fontSize": "9px",
                    "textAlign": "left",
                  }
                }
              >
                <li>
                  AP Computer Science
                </li>
                <li>
                  AP Computer Science Principles
                </li>
                <li>
                  AP Psychology
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginLeft": "5vw",
        "position": "relative",
        "top": "8vh",
        "width": "44vw",
      }
    }
  >
    <h2
      style={
        Object {
          "backgroundColor": "skyblue",
          "clear": "both",
          "float": "left",
          "fontFamily": "Georgia",
          "fontSize": "30px",
          "marginBottom": "3vh",
          "marginLeft": "1vw",
          "position": "relative",
        }
      }
    >
      <b>
        Experience
      </b>
    </h2>
    <div
      className="accordion"
    >
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": "35px",
            "flexDirection": "row",
            "float": "left",
            "marginBottom": "5vh",
            "maxHeight": "28vh",
            "minHeight": "22vh",
            "minWidth": "38vw",
          }
        }
      >
        <img
          className="card-img-left"
          src="llnllogo.jpeg"
          style={
            Object {
              "height": "73px",
              "marginLeft": "33px",
              "marginTop": "35px",
              "width": "73px",
            }
          }
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
            style={
              Object {
                "fontFamily": "Georgia",
                "fontSize": "20px",
              }
            }
          >
            <a
              href="https://www.llnl.gov/"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
              target="_blank"
            >
              <b>
                Lawrence Livermore 
                <br />
                National Laboratory
              </b>
            </a>
          </div>
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
                "marginLeft": "39px",
              }
            }
          >
            Livermore, CA   
            <b>
              |
            </b>
               Apr 2019 - Aug 2019
          </p>
          <br />
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
                "marginLeft": "39px",
              }
            }
          >
            Computation Student Scholar
          </p>
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <b>
            &gt;
          </b>
        </button>
        <div
          aria-expanded={null}
          className="collapse"
          style={
            Object {
              "maxWidth": "30vw",
            }
          }
        >
          <div
            className="card-body"
          >
            <div>
              <b>
                <u>
                  Responsibilities:
                </u>
              </b>
              <ul
                style={
                  Object {
                    "fontSize": "10px",
                    "textAlign": "left",
                  }
                }
              >
                <li>
                  Developed & designed a rationalization app for department directors at the lab, which consisted of a complex updateable ledger (table of accounting data) with the use of ag-Grid.
                </li>
                <li>
                  Implemented various system features for our admin pages, some of which included options to lock users out, post system messages, and give updates regarding the status of our finance application.
                </li>
                <li>
                  Updated and remodeled many of the finance department's SPAs (single-page applications) from AngularJS to Angular.
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginLeft": "5vw",
        "position": "relative",
        "top": "8vh",
        "width": "44vw",
      }
    }
  >
    <div
      className="accordion"
    >
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": "35px",
            "flexDirection": "row",
            "float": "left",
            "marginBottom": "5vh",
            "maxHeight": "28vh",
            "minHeight": "22vh",
            "minWidth": "38vw",
          }
        }
      >
        <img
          className="card-img-left"
          src="sfologo.png"
          style={
            Object {
              "height": "110px",
              "marginLeft": "20px",
              "marginRight": "10px",
              "marginTop": "15px",
              "width": "90px",
            }
          }
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
            style={
              Object {
                "fontFamily": "Georgia",
                "fontSize": "20px",
              }
            }
          >
            <a
              href="https://sfoconnect.com/"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
              target="_blank"
            >
              <b>
                City & County of San Francisco 
                <br />
                 (SFO Airport - ITT Office)
              </b>
            </a>
          </div>
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
                "marginLeft": "10px",
              }
            }
          >
            San Francisco, CA   
            <b>
              |
            </b>
               Jun 2017 - Aug 2017
          </p>
          <br />
          <p
            className="card-text"
            style={
              Object {
                "clear": "both",
                "float": "left",
                "fontFamily": "Raleway",
                "fontSize": "15px",
                "marginLeft": "10px",
              }
            }
          >
            Public Service Trainee (SWE Intern)
          </p>
        </div>
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <b>
            &gt;
          </b>
        </button>
        <div
          aria-expanded={null}
          className="collapse"
          style={
            Object {
              "maxWidth": "30vw",
            }
          }
        >
          <div
            className="card-body"
          >
            <div>
              <b>
                <u>
                  Responsibilities:
                </u>
              </b>
              <ul
                style={
                  Object {
                    "fontSize": "10px",
                    "textAlign": "left",
                  }
                }
              >
                <li>
                  Wrote unit tests in Java and ensured code coverage for the checkpoint wait time team with the use of Jenkins.
                </li>
                 
                <br />
                <li>
                  Collected Wi-Fi access point data and developed algorithms to determine the faulty or problematic access points.
                </li>
                 
                <br />
                <li>
                  Learned about RESTful APIs and the development of web services.
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
